### DISCUSSION TITLE
How to get image size from API route (hosted on Vercel)?

### DISCUSSION THREAD

**Discussion by cibulka** [vercel/community#850] [Help]
Hi, 

I need to get image dimensions from an API route of Next.js. Locally (both with `npm run dev` and `npm run build && npm start`) this (pseudo) code works:

```
import path from 'path';
import imageSize from 'image-size';

export default async function ApiEndpoint(req: NextApiRequest, res: NextApiResponse) {
  const { width, height } = imageSize(path.join('./public', photo));
  ... other things
}
```

On Vercel however it does not and I get this error:

```
Error: ENOENT: no such file or directory, open '/var/task/public/stories/tatiana.jpeg'
```

I've tried variations of the path (such as 'public', './public', '/public' or omitting "public") to no avail. 

Is there a recommended way how to read a file dynamically (not with `import`) in the API route of Next.js hosted on Vercel?

Thanks in advance!

**Comment by amyegan** [comment]
You'll need to use "process.cwd()" to get the current working directory. So you would have something like `path.join(process.cwd(), 'public', photo)`

**Reply by cibulka** [reply]
Hello @amyegan, thanks for your reply!

Unfortunately your suggestion does not help. The error remaíns the same:

```
Error: ENOENT: no such file or directory, open '/var/task/public/stories/tatiana.jpeg'
```
Do you want me to build a repo to reproduce the issue? 

Thank you, P!

**Reply by cibulka** [reply]
I've posted my question to SO as well: https://stackoverflow.com/questions/73712180/how-to-get-image-size-from-next-jss-api-route-hosted-on-vercel

**Reply by amyegan** [reply]
If you can share a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), then I would be happy to take a closer look into the source of the error.
